src/pages-and-resources/pages/PageCard.jsx

```diff
@@ -33,6 +33,31 @@ function PageCard({
     history.push(`${pagesAndResourcesPath}/${page.id}`);
   };
 
+  const settingsButton = () => {
+    if (page.legacyLink) {
+      return (
+        <Hyperlink destination={page.legacyLink}>
+          <Icon
+            className="mb-0 mr-1"
+            src={ForwardArrow}
+            size="inline"
+            screenReaderText={intl.formatMessage(messages.settings)}
+          />
+        </Hyperlink>
+      );
+    }
+    return (
+      <IconButton
+        className="mb-0 mr-1"
+        src={Settings}
+        iconAs={Icon}
+        size="inline"
+        alt={intl.formatMessage(messages.settings)}
+        onClick={() => history.push(`${pagesAndResourcesPath}/${page.id}/settings`)}
+      />
+    );
```

Comment:
what I suggest is to have a single return statement and make the logic inside the return like this 

`page.legacyLink ?  <Hyperlink destination={page.legacyLink}>  :  <IconButton>`

however what you have done is totally fine.